File: app/src/main/java/com/hxs/xposedreddevil/ui/MainActivity.java
package com.hxs.xposedreddevil.ui;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.Switch;
import android.widget.Toast;

import com.hxs.xposedreddevil.R;
import com.hxs.xposedreddevil.base.BaseActivity;
import com.hxs.xposedreddevil.databinding.ActivityHomeBinding;
import com.hxs.xposedreddevil.databinding.ActivityMainBinding;
import com.hxs.xposedreddevil.utils.MessageEvent;
import com.hxs.xposedreddevil.utils.MultiprocessSharedPreferences;
import com.hxs.xposedreddevil.utils.PushUtils;

import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

import androidx.annotation.RequiresApi;

public class MainActivity extends BaseActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(LayoutInflater.from(this));
        setContentView(binding.getRoot());
        EventBus.getDefault().register(this);
        SwitchClickInit();
    }

    private void SwitchClickInit() {
        try {
            if (sharedPreferences.getString("redmain", "2").equals("1")) {
                binding.swMain.setChecked(true);
            } else {
                binding.swMain.setChecked(false);
            }
            if (sharedPreferences.getString("red", "2").equals("1")) {
                binding.swOwn.setChecked(true);
            } else {
                binding.swOwn.setChecked(false);
            }
            if (sharedPreferences.getString("private", "2").equals("1")) {
                binding.swSlno.setChecked(true);
            } else {
                binding.swSlno.setChecked(false);
            }
            if (sharedPreferences.getString("sleep", "2").equals("1")) {
                binding.swSleep.setChecked(true);
                binding.llSleep.setVisibility(View.VISIBLE);
            } else {
                binding.swSleep.setChecked(false);
                binding.llSleep.setVisibility(View.GONE);
            }
            if (sharedPreferences.getString("sound", "2").equals("1")) {
                binding.swSound.setChecked(true);
            } else {
                binding.swSound.setChecked(false);
            }
            if (sharedPreferences.getString("push", "2").equals("1")) {
                binding.swPush.setChecked(true);
            } else {
                binding.swPush.setChecked(false);
            }
            if (sharedPreferences.getString("withdraw", "2").equals("1")) {
                binding.swWithdraw.setChecked(true);
            } else {
                binding.swWithdraw.setChecked(false);
            }
            if (sharedPreferences.getString("money", "2").equals("1")) {
                binding.swMoney.setChecked(true);
            } else {
                binding.swMoney.setChecked(false);
            }
            if (sharedPreferences.getString("openwechat", "2").equals("1")) {
                binding.swWechat.setChecked(true);
            } else {
                binding.swWechat.setChecked(false);
            }
            binding.etSleep.setText(sharedPreferences.getString("sleeptime", "1"));
        } catch (Exception e) {
            e.printStackTrace();
        }
        binding.swMain.setOnCheckedChangeListener((compoundButton, b) -> {
            if (b) {
                sharedPreferences.edit().putString("redmain", "1").commit();
            } else {
                sharedPreferences.edit().putString("redmain", "2").commit();
            }
        });
        binding.swOwn.setOnCheckedChangeListener((compoundButton, b) -> {
            if (b) {
                sharedPreferences.edit().putString("red", "1").commit();
            } else {
                sharedPreferences.edit().putString("red", "2").commit();
            }
        });
        //TODO 不抢私聊红包
        binding.swSlno.setOnCheckedChangeListener((compoundButton, b) -> {
            binding.swSlno.setChecked(false);
            Toast.makeText(this, "还在开发中~", Toast.LENGTH_SHORT).show();
            if (b) {
                sharedPreferences.edit().putString("private", "1").commit();
            } else {
                sharedPreferences.edit().putString("private", "2").commit();
            }
        });
        binding.swSleep.setOnCheckedChangeListener((compoundButton, b) -> {
            if (b) {
                sharedPreferences.edit().putString("sleep", "1").commit();
                binding.llSleep.setVisibility(View.VISIBLE);
            } else {
                sharedPreferences.edit().putString("sleep", "2").commit();
                binding.llSleep.setVisibility(View.GONE);
            }
        });
        binding.swSound.setOnCheckedChangeListener((compoundButton, b) -> {
            if (b) {
                sharedPreferences.edit().putString("sound", "1").commit();
            } else {
                sharedPreferences.edit().putString("sound", "2").commit();
            }
        });
        binding.swPush.setOnCheckedChangeListener((compoundButton, b) -> {
            if (b) {
                sharedPreferences.edit().putString("push", "1").commit();
            } else {
                sharedPreferences.edit().putString("push", "2").commit();
            }
        });
        binding.swWithdraw.setOnCheckedChangeListener((buttonView, b) -> {
            if (b) {
                sharedPreferences.edit().putString("withdraw", "1").commit();
            } else {
                sharedPreferences.edit().putString("withdraw", "2").commit();
            }
        });
        binding.swMoney.setOnCheckedChangeListener((buttonView, b) -> {
            if (b) {
                sharedPreferences.edit().putString("money", "1").commit();
            } else {
                sharedPreferences.edit().putString("money", "2").commit();
            }
        });
        binding.swWechat.setOnCheckedChangeListener((buttonView, b) -> {
            if (b) {
                sharedPreferences.edit().putString("openwechat", "1").commit();
            } else {
                sharedPreferences.edit().putString("openwechat", "2").commit();
            }
        });
        binding.etSleep.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {

            }

            @Override
            public void afterTextChanged(Editable editable) {
                if (editable.toString().equals("0")) {
                    sharedPreferences.edit().putString("sleeptime", "1").commit();
                } else {
                    sharedPreferences.edit().putString("sleeptime", editable.toString()).commit();
                }
            }
        });
        binding.llAddNo.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, SelectFilterActivity.class)));
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    @Subscribe(threadMode = ThreadMode.MAIN)
    public void getMsg(MessageEvent messages) {
        PushUtils.showNotification(this, "private", "25", "微信", "天降红包");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
    }

}

File: app/src/main/java/com/hxs/xposedreddevil/ui/MainActivity.kt
package com.hxs.xposedreddevil.ui

import android.content.Intent
import android.content.SharedPreferences
import android.os.Build
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.widget.CompoundButton
import android.widget.Toast
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AlertDialog
import com.hxs.xposedreddevil.base.BaseActivity
import com.hxs.xposedreddevil.databinding.ActivityMainBinding
import com.hxs.xposedreddevil.utils.MessageEvent
import com.hxs.xposedreddevil.utils.PushUtils
import com.hxs.xposedreddevil.utils.SuUtil
import com.hxs.xposedreddevil.utils.xshare.Utils
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode
import kotlin.system.exitProcess

class MainActivity : BaseActivity() {
    private var binding: ActivityMainBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(LayoutInflater.from(this))
        setContentView(binding!!.root)
        EventBus.getDefault().register(this)
        SwitchClickInit()
    }

    private fun SwitchClickInit() {
        try {
            binding!!.swMain.isChecked = config.redMain
            binding!!.swOwn.isChecked = config.red
            binding!!.swSlno.isChecked = config.privates
            if (config.sleep) {
                binding!!.swSleep.isChecked = true
                binding!!.llSleep.visibility = View.VISIBLE
            } else {
                binding!!.swSleep.isChecked = false
                binding!!.llSleep.visibility = View.GONE
            }
            binding!!.swSound.isChecked = config.sound
            binding!!.swPush.isChecked = config.push
            binding!!.swWithdraw.isChecked = config.withdraw
            binding!!.swMoney.isChecked = config.money
            binding!!.swWechat.isChecked = config.openwechat
            binding!!.etSleep.setText(config.sleeptime)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        binding!!.swMain.setOnCheckedChangeListener { compoundButton: CompoundButton?, b: Boolean ->
            config.redMain = b
            SuUtil.kill("com.tencent.mm")
        }
        binding!!.swOwn.setOnCheckedChangeListener { compoundButton: CompoundButton?, b: Boolean ->
            config.red = b
            SuUtil.kill("com.tencent.mm")
        }
        //TODO 不抢私聊红包
        binding!!.swSlno.setOnCheckedChangeListener { compoundButton: CompoundButton?, b: Boolean ->
            binding!!.swSlno.isChecked = false
            Toast.makeText(this, "还在开发中~", Toast.LENGTH_SHORT).show()
            config.privates = b
        }
        binding!!.swSleep.setOnCheckedChangeListener { compoundButton: CompoundButton?, b: Boolean ->
            config.sleep = b
            if (b) {
                binding!!.llSleep.visibility = View.VISIBLE
            } else {
                binding!!.llSleep.visibility = View.GONE
            }
            SuUtil.kill("com.tencent.mm")
        }
        binding!!.swSound.setOnCheckedChangeListener { compoundButton: CompoundButton?, b: Boolean ->
            config.sound = b
            SuUtil.kill("com.tencent.mm")
        }
        binding!!.swPush.setOnCheckedChangeListener { compoundButton: CompoundButton?, b: Boolean ->
            config.push = b
            SuUtil.kill("com.tencent.mm")
        }
        binding!!.swWithdraw.setOnCheckedChangeListener { buttonView: CompoundButton?, b: Boolean ->
            config.withdraw = b
            SuUtil.kill("com.tencent.mm")
        }
        binding!!.swMoney.setOnCheckedChangeListener { buttonView: CompoundButton?, b: Boolean ->
            config.money = b
            SuUtil.kill("com.tencent.mm")
        }
        binding!!.swWechat.setOnCheckedChangeListener { buttonView: CompoundButton?, b: Boolean ->
            config.openwechat = b
            SuUtil.kill("com.tencent.mm")
        }
        binding!!.etSleep.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
            override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
            override fun afterTextChanged(editable: Editable) {
                config.sleeptime = if (editable.toString() == "0") "1" else editable.toString()
            }
        })
        binding!!.llAddNo.setOnClickListener { v: View? ->
            startActivity(
                Intent(
                    this@MainActivity,
                    SelectFilterActivity::class.java
                )
            )
        }
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    @Subscribe(threadMode = ThreadMode.MAIN)
    fun getMsg(messages: MessageEvent?) {
        PushUtils.showNotification(this, "private", "25", "微信", "天降红包")
    }

    override fun onRestart() {
        super.onRestart()
    }
}
